### PR TITLE
add Lab7

### PR DESCRIPTION
**Pull request by nqthanh1**
add Lab7, prob3 has been done

### CODE REVIEW

File: src/Lab7/prob3/ForEachExample.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	public static void main(String[] args) {
+		List<String> list = Arrays.asList("Hello there", "Goodbye", "Back soon", 
+				"Away", "On Vacation", "Everywhere you want to be");
+		
+		//print each element of the list in upper case format
+		Consumer<String> upperCaseName = name -> System.out.println(toUpper(name));
+		list.forEach(upperCaseName);
+
+		//implements a consumer class
+		list.forEach(new PrintUpperCaseConsumer());
```
Comment (vietnguyen) [inline src/Lab7/prob3/ForEachExample.java:19]: What is this line for ? I think we can remove it.

File: src/Lab7/prob3/ForEachExample.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+		// Inline Consumer implementation
+		Consumer<String> printUpperCase = new Consumer<>() {
+			@Override
+			public void accept(String s) {
+				System.out.println(s.toUpperCase());
+			}
+		};
+
+		// Use forEach with the inline Consumer
+		list.forEach(printUpperCase);
```
Comment (vietnguyen) [inline src/Lab7/prob3/ForEachExample.java:30]: Need create an instance of class `PrintUpperCaseConsumer` then pass the List item into the Consumer.
```
PrintUpperCaseConsummer consumer = new PrintUpperCaseConsumer();
list.forEach(item -> consumer.accept(item));
```

Comment (vietnguyen) [inline src/Lab7/prob3/ForEachExample.java:30]: Need create an instance of class `PrintUpperCaseConsumer` then pass the List item into the Consumer.
```
PrintUpperCaseConsummer consumer = new PrintUpperCaseConsumer();
list.forEach(item -> consumer.accept(item));
```

File: src/Lab7/prob3/ForEachExample.java (hunk 1 above)
Comment (nqthanh1) [reply src/Lab7/prob3/ForEachExample.java:19]: three ways of using Consumer

**Review by vietnguyen — CHANGES_REQUESTED** [review]
Please fix prob3

File: src/Lab7/prob3/ForEachExample.java (hunk 2 above)
Comment (nqthanh1) [reply src/Lab7/prob3/ForEachExample.java:30]: already override the accept method so it should be called automatically. Tested

File: src/Lab7/prob3/ForEachExample.java (hunk 1 above)
Comment (vietnguyen) [reply src/Lab7/prob3/ForEachExample.java:19]: that line of code just create new instance of `PrintUpperCaseConsumer` in each loop. Try to test by running the `main` method and verify if it prints anything to the console

**Comment by nqthanh1** [comment]
> Please fix prob3

There are three different ways to implement the Consumer inferface in Prob3

**Review by vietnguyen — APPROVED** [review]

**Pull request merged**